src/main.cpp

```diff
@@ -270,18 +284,34 @@ void returnToMainMenu(RenderLayer* render_layer)
         return;
     }
 
-    if (PreferencesManager::get("headless") != "")
+    string headless = PreferencesManager::get("headless", "");
+    if (!headless.empty())
     {
-        new EpsilonServer(defaultServerPort);
+        // Create the server to listen on the assigned port.
+        // Use the default port if server_port isn't set or has an invalid
+        // value (toInt returns 0).
+        int headless_port = PreferencesManager::get("server_port").toInt();
+        // This is the same process as server_port and could be made DRY.
+        if (headless_port < 10 || headless_port > 65535)
+        {
+            new EpsilonServer(defaultServerPort);
```

Comment:
Instead of doing `new EpsilonServer` in both the if and else, I find it easier to read:
```
if (headless_port < 10 || headless_port > 65535) {
  LOG(Warning, ...);
  headless_port = default_port;
}
new Server(..);
```